### PR TITLE
Feature enumerables

### PR DESCRIPTION
**Pull request by pipe2442**
Enumerables methods is a project in wich we had to recreate some enumerable methods to understand the functionality of Ruby methods
Recreated Methods:
- each
- each_with_index
- select
- all?
- any?
- none?
- count
- map 
- inject

### CODE REVIEW

**Comment by davidmrtz-dev** [comment]
## Hi Team. 👨‍🚀
### Very good job so far.

However, there are some issues that need to be fixed, please check the comments below.  👨‍💻  ⬇️ 

****

- [x]  Pull Request:
Your PR is good, but you can improve adding a shortlist with the methods you're recreating. 😄 

****
**Please make the changes required and submit for re-review once you are ready!** 
****
### Status: Changes required. ♻️  
****


**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,332 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/Void
+p '-----------|||||--------'
+p '----------{(o o)}--------'
+p '--------ooO-(_)-Ooo------'
+# Initial comment
+module Enumerable
+  def my_each
```
Comment (davidmrtz-dev) [inline enumerables.rb:7]: My_each when a block is given and when it is an array, should return the array itself after calling the given block once for each element in the array.

Screen Shot 2020-07-21 at 7 16 41 PM

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index]) }
+  end
+
+  def my_each_with_index
```
Comment (davidmrtz-dev) [inline enumerables.rb:15]: When a block is given and it is a range, My each with index should return the range itself after calling the given block once for each element in self.
Screen Shot 2020-07-21 at 7 21 55 PM

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    array = to_a
+    arraya = []
+    array.length.times do |a|
+      array[a]
+      arraya << array[a] if yield (array[a])
+    end
+    arraya
+  end
+
+  def my_all?(*arg)
```
Comment (davidmrtz-dev) [inline enumerables.rb:35]: When no block or argument is given returns true when none of the collection members are false or nil.
Screen Shot 2020-07-21 at 7 24 31 PM
****
When no block or argument is given returns false when at least one of the collection members is false or nil.
Screen Shot 2020-07-21 at 7 26 30 PM
****
When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
Also returns false if any of the collection does not match the pattern as shown below.
Screen Shot 2020-07-21 at 7 28 50 PM
****

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+          next
+        end
+        return false
+      end
+    end
+  end
+end
+
+module Enumerable
+  def my_any?(*arg)
```
Comment (davidmrtz-dev) [inline enumerables.rb:99]: Should returns false if the block ever returns a value false or nil.
Screen Shot 2020-07-21 at 7 33 07 PM
****
When no block or argument is given returns true if at least one of the collection is not false or nil.
Screen Shot 2020-07-21 at 7 34 01 PM
****
When no block or argument is given returns false if at least one of the collection is not true.
Screen Shot 2020-07-21 at 7 35 26 PM
****
When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
Screen Shot 2020-07-21 at 7 37 48 PM
****
When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
Screen Shot 2020-07-21 at 7 38 38 PM

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+      array3.length.times do |b|
+        return true if arg[0].match(array3[b])
+
+        return false
+      end
+    end
+  end
+
+  def my_none?(*arg)
```
Comment (davidmrtz-dev) [inline enumerables.rb:142]: Should returns false if the block ever returns true for all elements::range
Screen Shot 2020-07-21 at 7 48 10 PM
****
When no block or argument is given returns true only if none of the collection members is true.
Screen Shot 2020-07-21 at 7 51 22 PM
****
When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
Screen Shot 2020-07-21 at 7 53 13 PM

File: enumerables.rb (hunk 3 above)
Comment (Alexoid1) [reply enumerables.rb:33]: ## specify pattern
Traceback (most recent call last):
enumerables.rb:364:in `<main>': undefined local variable or method `arr' for main:Object (NameError)
Did you mean?  Array

File: enumerables.rb (hunk 5 above)
Comment (Alexoid1) [reply enumerables.rb:142]: ## specify argument &true_block return and error also with the original method
code

Traceback (most recent call last):
enumerables.rb:400:in `<main>': undefined local variable or method `true_block' for main:Object (NameError)

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, great job with this project :+1: PR :100: README :100: Congratulation on fixing all the comments made by the previous TSE :1st_place_medal: Now the project meets all the minimum requirements :fireworks: 

Go ahead and merge to development branch :slightly_smiling_face: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**